module_utils/network_lsr/argument_validator.py

```diff
@@ -698,6 +699,64 @@ def _validate_post(self, value, name, result):
         return result
 
 
+class ArgValidatorPath(ArgValidatorStr):
+    """
+    Valides that the value is a valid posix absolute path
+    """
+
+    def __init__(self, name, required=False, default_value=None):
+        ArgValidatorStr.__init__(self, name, required, default_value, None)
+
+    def _validate_impl(self, value, name):
+        ArgValidatorStr._validate_impl(self, value, name)
+
+        if posixpath.isabs(value) is False:
+            raise ValidationError(
+                name, "value '%s' is not a valid posix absolute path" % (value),
+            )
+        return value
+
+
+class ArgValidator_Dict802_1X(ArgValidatorDict):
```

Comment:
I think a 802-1x setting only works for type `wifi` and `ethernet`. That is, because it's only implemented for those types. In NetworkManager you cannot do 802.1x port authentication on (for example) a bridge interface.

I think `_validate_post()` should reject specifying a 802.1x setting for unsupported types.

Reply:
Will add this to the _validate_post

Reply:
Yeah, I thought that was a better spot to put it as it is easy to determine if the role is being run with NetworkManager or initscripts